src/Adapter/Category/QueryHandler/GetCategoryForEditingHandler.php

```diff
@@ -66,7 +67,7 @@ public function handle(GetCategoryForEditing $query)
     {
         $category = new Category($query->getCategoryId()->getValue());
 
-        if (!$category->id) {
+        if (!$category->id || (!$category->isAssociatedToShop() && Shop::getContext() == Shop::CONTEXT_SHOP)) {
```

Comment:
The condition is a bit different, so maybe a dedicated exception should be thrown?
Like `InvalidShopContextException` ? or a very explicit `CategoryNotFoundInContextException`

Reply:
In fact it's the original behaviour for `CategoryNotFoundException` 😅

Reply:
Ok, I won't question this logic then 😛